networkx/linalg/algebraicconnectivity.py

```diff
@@ -244,7 +244,7 @@ def P(x):
             W -= (W.T * X * X.T).T
             project(W)
             # Compute the diagonal of P * L * P as a Jacobi preconditioner.
-            D = L.diagonal()
+            D = L.diagonal().astype(float)
             D += 2. * (asarray(X) * asarray(W)).sum(axis=1)
```

Comment:
Thanks @ysitu for helping out here...

as usual I still don't understand... ???The D used in line 205 is presumably set in line 174. But if not set in line 174, then it is searched for in the global namespace??? And this is a lambda definition, so D is only evaluated once at line 205 and the lambda function is not updated when D is redefined below.

As far as I can tell ```_tracemin_fiedler``` is not recursive in any way, so the stuff we are doing at line 247 doesn't affect the lambda function sent to the PCG solver. Maybe I don't understand how lambda's work, but it seems like updating the variable called ```D``` does not impact the PCG solver at all...???...